src/Sulu/Component/Media/SystemCollections/SystemCollectionManager.php

```
@@ -0,0 +1,262 @@
+                $key = sprintf('%s.%s', $namespaceKey, $collectionKey);
+                $collections[$key] = $this->getOrCreateCollection(
+                    $key,
+                    $collectionItem['meta_title'],
+                    $userId,
+                    $namespace->getId()
+                )->getId();
+            }
+        }
+
+        $this->entityManager->flush();
+
+        return $collections;
+    }
+
+    /**
+     * Finds or create a new system-collection namespace.
+     *
+     * @param string $namespace
+     * @param string $title
+     * @param string $locale
+     * @param int $userId
+     * @param int|null $parent id of parent collection or null for root.
+     *
+     * @return Collection
+     */
+    private function getOrCreateRoot($namespace, $title, $locale, $userId, $parent = null)
+    {
+        if (($collection = $this->collectionManager->getByKey($namespace, $locale)) !== null) {
+            $collection->setTitle($title);
+
+            return $collection;
+        }
+
+        return $this->createCollection($title, $namespace, $locale, $userId, $parent);
+    }
+
+    /**
+     * Finds or create a new system-collection.
+     *
+     * @param string $key
+     * @param array $localizedTitles
+     * @param int $userId
+     * @param int|null $parent id of parent collection or null for root.
+     *
+     * @return Collection
+     */
+    private function getOrCreateCollection($key, $localizedTitles, $userId, $parent)
+    {
+        $locales = array_keys($localizedTitles);
+        $firstLocale = array_shift($locales);
+
+        $collection = $this->collectionManager->getByKey($key, $firstLocale);
+        if ($collection === null) {
+            $collection = $this->createCollection($localizedTitles[$firstLocale], $key, $firstLocale, $userId, $parent);
+        } else {
+            $collection->setTitle($localizedTitles[$firstLocale]);
```

Just an idea: Wouldn't it be better to throw an exception here, instead of just updating a collection you do not know anything about?